cmd/thanos/query-frontend.go

```diff
@@ -10,99 +10,79 @@ import (
 	"github.com/NYTimes/gziphandler"
 	"github.com/cortexproject/cortex/pkg/querier/frontend"
 	"github.com/cortexproject/cortex/pkg/querier/queryrange"
+	"github.com/cortexproject/cortex/pkg/util/validation"
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
 	"github.com/oklog/run"
 	"github.com/opentracing/opentracing-go"
 	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/common/model"
-	"github.com/thanos-io/thanos/pkg/extkingpin"
-	"github.com/weaveworks/common/user"
-
 	"github.com/thanos-io/thanos/pkg/component"
 	"github.com/thanos-io/thanos/pkg/extflag"
+	"github.com/thanos-io/thanos/pkg/extkingpin"
 	"github.com/thanos-io/thanos/pkg/extprom"
 	extpromhttp "github.com/thanos-io/thanos/pkg/extprom/http"
 	"github.com/thanos-io/thanos/pkg/logging"
 	"github.com/thanos-io/thanos/pkg/prober"
 	"github.com/thanos-io/thanos/pkg/queryfrontend"
-	"github.com/thanos-io/thanos/pkg/queryfrontend/cache"
 	httpserver "github.com/thanos-io/thanos/pkg/server/http"
 	"github.com/thanos-io/thanos/pkg/server/http/middleware"
 	"github.com/thanos-io/thanos/pkg/tracing"
+	"github.com/weaveworks/common/user"
+	"gopkg.in/yaml.v2"
 )
 
-type queryFrontendConfig struct {
-	http             httpConfig
-	queryRangeConfig queryRangeConfig
-
-	downstreamURL        string
-	compressResponses    bool
-	LogQueriesLongerThan time.Duration
+type config struct {
+	http httpConfig
+	queryfrontend.Config
 
+	cachePathOrContent     extflag.PathOrContent
 	requestLoggingDecision string
-}
-
-type queryRangeConfig struct {
-	respCacheConfig     extflag.PathOrContent
-	cacheMaxFreshness   time.Duration
-	splitInterval       model.Duration
-	maxRetries          int
-	maxQueryParallelism int
-	maxQueryLength      model.Duration
-
 	// partialResponseStrategy is the default strategy used
 	// when parsing thanos query request.
 	partialResponseStrategy bool
 }
 
-func (c *queryRangeConfig) registerFlag(cmd extkingpin.FlagClause) {
-	cmd.Flag("query-range.split-interval", "Split queries by an interval and execute in parallel, 0 disables it.").
-		Default("24h").SetValue(&c.splitInterval)
+func registerQueryFrontend(app *extkingpin.App) {
+	comp := component.QueryFrontend
```

Comment:
Why are we changing all these flag names? What's the motivation behind it?

Reply:
because I think it makes sense to use exactly the same names as in Cortex

Reply:
WDYT? @yeya24 What was your motivation when you first changed these?

Reply:
Also because of these changes. We have to update kube-thanos as well. My vote on keeping them as is. Unless we want to make thanos query frontend a drop-in replacement for cortex frontend, which I don't see any value. We are already sharing a common codebase and if we can implement something on upstream we'll be implementing it.

Reply:
aha yeah good point about kube-thanos, but OTOH `query-frontend.log-queries-longer-than` needs changing anyway as it was with underscores so kube-thanos needs to be changed anyway.

Reply:
I named these flags based on the original proposal https://thanos.io/tip/proposals/202004_embedd_cortex_frontend.md/. 

I agree with @kakkoyun it is not necessary to be the same as Cortex flags and to do this we have to change a lot.

Reply:
> Sure, but how flag naming changes anything? Still, I don't get the changes to those flags in some cases. The underscore -> hyphen makes sense but some naming was adjusted for readability reasons.

ooh yes you are right only the config struct passes through some validation.